src/main/kotlin/cacophony/semantic/syntaxtree/SyntaxTree.kt

```diff
@@ -4,9 +4,19 @@ import cacophony.utils.Location
 import cacophony.utils.Tree
 import cacophony.utils.TreeLeaf
 
-sealed interface LeafExpression
+sealed interface SyntaxTree {
+    val range: Pair<Location, Location>
 
-sealed interface SyntaxTree
+    fun isEquivalent(other: SyntaxTree?): Boolean = other != null && range == other.range
+}
+
+sealed interface Expression : SyntaxTree, Tree
+
+sealed interface Assignable : Expression
```

Comment:
Are there cases in which Assignable is not a TreeLeaf?

Reply:
as I unterstand this, FieldRef corresponding to `s.x` is assignable (if s is also assignable) and it has s as its child

Reply:
Yep, `FieldRef` is not a `TreeLeaf`

Reply:
fair